### PR TITLE
basecamps: use autosort zones for menial sort and food distribution

### PR DESCRIPTION
**Pull request by mlangsdorf**
#### Summary
```SUMMARY: Features "basecamps: use autosort zones for menial sort and food distribution"```

#### Purpose of change
Get rid of the weird and confusing sort point mechanic, and replace it with the slightly better supported zone loot sorting mechanic.  Adjust the Food Distribution pseudo-mission to work with the new loot zones.

#### Describe the solution
Add a new function to zone_manager:: that checks if there are any loot destination zones nearby.

Pull most of the logic from do_turn_move_loot into menial_return, except for the bits that actually move the player around.  Execute that code to sort the loot.

#### Describe alternatives you've considered
I have an effort to get NPCs to use sort zones and auto-sort with #28571, but it's hard.  Getting rid of sort points helps with the effort to get rid of the camp overseer and use the faction manager to control basecamps.

### CODE REVIEW

File: src/faction_camp.cpp [hunk 1]
```diff
@@ -3113,59 +3166,24 @@ std::vector<item *> basecamp::give_equipment( std::vector<item *> equipment,
...
-    if( query_yn( show_pts ) ) {
-        sort_points.clear();
-        sort_points.insert( sort_points.end(), new_pts.begin(), new_pts.end() );
-        return true;
-    }
-    if( query_yn( _( "Revert to default points?" ) ) ) {
-        return set_sort_points( true, false );
-    }
-    return false;
+    popup( _( "Sorting zones have changed.  Please create some sorting zones.  "
```
Comment (kevingranade) [inline src/faction_camp.cpp]: Looks like you need a Basecamp: Food, a Loot: anything, and Loot: Unsorted
At first I couldn't figure out what it wanted, then after checking the source I got stuck trying to add them, and I'm not clear why.
tl;dr please make the message explicit, and either fail out of the menu, or give an option to exit it.

Comment (mlangsdorf) [reply src/faction_camp.cpp]: Good point.  Message made more explicit, and an option to not set sort points added.

File: src/faction_camp.cpp [hunk 2]
```diff
@@ -649,9 +637,9 @@ void basecamp::get_available_missions( mission_data &mission_key )
 
         entry = string_format( _( "Notes:\n"
                                   "Distribute food to your follower and fill you larders.  Place "
-                                  "the food you wish to distribute on the companion food sort "
-                                  "point.  By default, that is opposite the tent door between "
-                                  "the manager and wall.\n \n"
+                                  "the food you wish to distribute in the camp food zone.  You "
+                                  "must create a camp food zonei near the camp using the zone "
```
Comment (kevingranade) [inline src/faction_camp.cpp]:
```suggestion
                                  "must create a camp food zone near the camp using the zone "
```

File: src/faction_camp.cpp [hunk 3]
```diff
@@ -2087,73 +2074,138 @@ bool basecamp::menial_return()
...
+    if( !mgr.is_sorting() ) {
+        mgr.start_sort( src_sorted );
     }
-    //Undo our hack!
-    for( tripoint &sort_point : sort_points ) {
-        if( g->m.furn( sort_point ) == f_ash ) {
-            g->m.furn_set( sort_point, f_null );
+    for( auto &src : src_sorted ) {
+        const auto &src_loc = g->m.getlocal( src );
+        // skip tiles in IGNORE zone and tiles on fire (to prevent taking out wood off thei
```
Comment (kevingranade) [inline src/faction_camp.cpp]:
```suggestion
        // skip tiles in IGNORE zone and tiles on fire (to prevent taking out wood off the
```

**Comment by mlangsdorf** [comment]
#28571 has a better solution that is 95% working, so closing this.
